config.go

```diff
@@ -0,0 +1,213 @@
+	Development bool `json:"development",yaml:"development"`
+	// DisableCaller stops annotating logs with the calling function's file
+	// name and line number.
+	DisableCaller bool `json:"disable_caller",yaml:"disable_caller"`
+	// DisableStacktrace completely disables automatic stacktrace capturing.
+	DisableStacktrace bool `json:"disable_stacktrace",yaml:"disable_stacktrace"`
+	// Sampling sets a sampling policy. A nil SamplingConfig disables sampling.
+	Sampling *SamplingConfig `json:"sampling",yaml:"sampling"`
+	// Encoding sets the logger's encoding. Valid values are "json" and
+	// "console".
+	Encoding string `json:"encoding",yaml:"encoding"`
+	// EncoderConfig sets options for the chosen encoder. See
+	// zapcore.EncoderConfig for details.
+	EncoderConfig zapcore.EncoderConfig `json:"encoder_config",yaml:"encoder_config"`
+	// OutputPaths is a list of paths to write logging output to. See Open for
+	// details.
+	OutputPaths []string `json:"output_paths",yaml:"output_paths"`
+	// ErrorOutputPaths is a list of paths to write internal logger errors to.
+	// The default is standard error.
+	ErrorOutputPaths []string `json:"error_output_paths",yaml:"error_output_paths"`
+	// InitialFields is a collection of fields to add to the root logger.
+	InitialFields map[string]interface{} `json:"initial_fields",yaml:"initial_fields"`
+}
+
+// NewProductionConfig is the recommended production configuration. Logging is
+// enabled at InfoLevel and above.
+//
+// It uses a JSON encoder, writes logs to stdout and internal errors to stderr,
+// and enables sampling. Stacktraces are automatically included on logs of
+// ErrorLevel and above.
+func NewProductionConfig() Config {
+	return Config{
+		Level:       DynamicLevel(),
+		Development: false,
+		Sampling: &SamplingConfig{
+			Initial:    100,
+			Thereafter: 100,
+		},
+		Encoding: "json",
+		EncoderConfig: zapcore.EncoderConfig{
+			TimeKey:        "ts",
+			LevelKey:       "level",
+			NameKey:        "logger",
+			CallerKey:      "caller",
+			MessageKey:     "msg",
+			StacktraceKey:  "stacktrace",
+			EncodeLevel:    zapcore.LowercaseLevelEncoder,
+			EncodeTime:     zapcore.EpochTimeEncoder,
+			EncodeDuration: zapcore.SecondsDurationEncoder,
+		},
+		OutputPaths:      []string{"stdout"},
+		ErrorOutputPaths: []string{"stderr"},
+	}
+}
+
+// NewDevelopmentConfig is a reasonable development configuration. Logging is
+// enabled at DebugLevel and above.
+//
+// It (obviously) enables development mode, uses a console encoder, writes logs
+// to stdout and internal errors to stderr, and disables sampling. Stacktraces
+// are automatically included on logs of WarnLevel and above.
+func NewDevelopmentConfig() Config {
+	dyn := DynamicLevel()
+	dyn.SetLevel(DebugLevel)
+
+	return Config{
+		Level:       dyn,
+		Development: true,
+		Encoding:    "console",
+		EncoderConfig: zapcore.EncoderConfig{
+			// Keys can be anything except the empty string.
+			TimeKey:        "T",
```

Comment:
The console encoder only cares about empty/non-empty - the actual values aren't used.

Terminology varies, but I'd call anything that's "deployed" a production config. In production, you may have canary, staging, and stable, but it's all production. This is for use while developers are writing code, running unit tests, and exploring on their local machines.